sui_programmability/examples/games_with_chance/sources/satoshi_flip.move

```diff
@@ -0,0 +1,303 @@
+                secret: msg,
+                guess: 2,
+                player_won: false
+            };
+            option::fill(&mut game.outcome, outcome);
+
+            // no bet placed, return the balance to house
+            let HouseData {house_balance, min_bet: _, max_bet: _} = option::extract(&mut game.house_data);
+            let house_coins = coin::from_balance(house_balance, ctx);
+            transfer::transfer(house_coins, game.house);
+        } else {
+            let hash = sha3_256(secret);
+            if (hash != digest::sha3_256_digest_to_bytes(&game.hashed_secret)) {
+                let BetData {stake, guess:_} = option::extract(&mut game.bet_data);
+                let HouseData {house_balance, min_bet: _, max_bet:_} = option::extract(&mut game.house_data);
+                let player = option::extract(&mut game.player);
+                // TODO: Maybe add some punishment for the house
+                transfer::transfer(coin::from_balance(stake, ctx), player);
+                transfer::transfer(coin::from_balance(house_balance, ctx), game.house);
+                abort EHashAndSecretDontMatch
+            };
+            // extract balances and guess
+            let HouseData {house_balance, min_bet: _, max_bet: _} = option::extract(&mut game.house_data);
+            let BetData {stake, guess} = option::extract(&mut game.bet_data);
+            let first_byte = vector::borrow(&secret, 0);
+            let won = guess == *first_byte % 2;
+
+            if (won) {
+                let outcome = Outcome {
+                    secret,
+                    guess,
+                    player_won: true
+                };
+                option::fill(&mut game.outcome, outcome);
+                let player = option::extract(&mut game.player);
+                pay_player(player, game.house, stake, house_balance, ctx);
+            } else {
+                // player lost
+                let outcome = Outcome {
+                    secret,
+                    guess,
+                    player_won: false
+                };
+                option::fill(&mut game.outcome, outcome);
+
+                balance::join(&mut house_balance, stake);
+                transfer::transfer(coin::from_balance(house_balance, ctx), game.house);
+            };
+        }
+    }
+
+    /// Called by anyone after the required epochs have passed and house has not revealed the secret.
+    /// The house automatically loses.
+    /// Check EpochsCancelAfter for the number of epochs required to pass after `fun bet` has been called
+    public entry fun cancel_game(game: &mut Game, ctx: &mut TxContext) {
+        // this can't be called on an ended game
+        assert!(option::is_none<Outcome>(&game.outcome), EGameAlreadyEnded);
+        // a bet has to have been placed
+        assert!(option::is_some<BetData>(&game.bet_data), ECannotCancelBeforeBetting);
+        // this call can only be called 2 epochs after the bet has been placed
```

Comment:
This comment is wrong (and generally when we refer to 2 epochs in the comments). Ie now we set it to 5, so ideally just replace number `2` with `EpochsCancelAfter`